### PR TITLE
[PROMPT]: Concatenate slash to cwd_dir if it doesn't end with a slash.

### PR DESCRIPTION
**Pull request by Eadaen**
Implements change to $PROMPT_FIELDS["cwd_dir"] functionality as proposed in #3751. Now cwd_dir consistently ends with a slash, to ease usability.

Previous behaviour(cwd -> cwd_dir): "/foo/bar" -> "/foo"; "~/foo" -> "~".
New behaviour: "/foo/bar" -> "/foo/"; "~/foo" -> "~/".
Behaviour if cwd is one of "~", "/" or "/foo" is unchanged

The code is cross-platform, though has only been tested on Unix-like paths.

Let me know if I should add tests for the function added, and where they should be added if so. tests/test_prompt.py?

### CODE REVIEW

File: xonsh/prompt/cwd.py [hunk 1]
```diff
@@ -89,3 +89,12 @@ def _dynamically_collapsed_pwd():
                 0 : int(target_width) - len(truncature_char)
             ] + truncature_char
     return full
+
+
+def _add_dir_slash(cwd_dir):
```
Comment (laloch) [inline xonsh/prompt/cwd.py:94]: I think this is usually implemented using `os.path.join(cwd_dir, "")`. No extra function needed.

**Review by laloch — CHANGES_REQUESTED** [review]
Hi @Eadaen, thanks for the PR. Could you please check my comment above?

File: xonsh/prompt/cwd.py (hunk 1 above)
Comment (Eadaen) [reply xonsh/prompt/cwd.py]: Thanks for the suggestion. That makes things simpler.

**Review by laloch — APPROVED** [review]

**Comment by laloch** [comment]
Thanks @Eadaen!

**Pull request merged**